Sources/Build/BuildDelegate.swift

```diff
@@ -186,6 +186,183 @@ extension SPMLLBuild.Diagnostic: DiagnosticDataConvertible {
     }
 }
 
+class CustomLLBuildCommand: ExternalCommand {
+    let ctx: BuildExecutionContext
+
+    required init(_ ctx: BuildExecutionContext) {
+        self.ctx = ctx
+    }
+
+    func getSignature(_ command: SPMLLBuild.Command) -> [UInt8] {
+        return []
+    }
+
+    func execute(_ command: SPMLLBuild.Command) -> Bool {
+        fatalError("subclass responsibility")
+    }
+}
+
+final class TestDiscoveryCommand: CustomLLBuildCommand {
```

Comment:
Docs would be nice, I have no idea what this command is trying to do :)

Reply:
Going to add some docs in a separate commit.